### ISSUE TITLE
Lack of parity between CLI values and UI answers when upgrading application

### ISSUE THREAD

**Issue by Jakedavies** [rancher/rancher#23410]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create an application: I don't think this part matters too much anything should work but mine is a helm chart.
- paste in some values as yaml via the editor in the UI.
- View the app in API (the values here will be saved as a raw escaped string of the yaml you input)
- Now upgrade the app via the CLI and pass a file with the same values you pasted in to the UI. `rancher app upgrade --values values.yaml appName version`
- View the app in the API again, the values.yaml you used will be populated into the answers section of the application spec instead of the valuesYaml section it was in previously. 

**Result:**

My application can be upgraded from the UI safely if pasting the yaml into the editor, but it breaks when upgrading from the CLI.

**Other details that may be helpful:**

**Environment information**
- Rancher version: : 2.2.8
- rancher cli version: 2.3.0

**Comment by deniseschannon** [comment]
@Jakedavies Typically, CLI versions are not backwards compatible as there are specific updates in Rancher server that need to be updated as well.

Are you facing this issue when running Rancher v2.3.0 and Rancher CLI v2.3.0?

**Comment by Jakedavies** [comment]
This was when running the rancher CLI version downloaded from my rancher installation. Rancher CLI 2.2 on rancher 2.2.8

**Comment by brendarearden** [comment]
available for testing in rancher/cli v2.4.3-rc1

**Comment by soumyalj** [comment]
Reproduced the issue with Rancher v2.2.8 using CLI 2.3.0
Created a DO cluster, deployed a datadog app with version v1.33.0. The app deployed successfully through UI. The API displayed values.yml as below:
```
"name": "datadog",
"namespaceId": null,
"notes": "---\n# Source: datadog/charts/kube-state-metrics/templates/NOTES.txt\nkube-state-metrics is a simple service that listens to the Kubernetes API server and generates metrics about the state of the objects.\nThe exposed metrics can be found here:\nhttps://github.com/kubernetes/kube-state-metrics/blob/master/docs/README.md#exposed-metrics\n\nThe metrics are exported on the HTTP endpoint /metrics on the listening port.\nIn your case, datadog-kube-state-metrics.datadog.svc.cluster.local:8080/metrics\n\nThey are served either as plaintext or protobuf depending on the Accept header.\nThey are designed to be consumed either by Prometheus itself or by a scraper that is compatible with scraping a Prometheus client endpoint.\n\n\n---\n# Source: datadog/templates/NOTES.txt\n\nDataDog agents are spinning up on each node in your cluster. After a few\nminutes, you should see your agents starting in your event stream:\n\n    https://app.datadoghq.com/event/stream\n\n",
"projectId": "c-v5l2s:p-5p794",
"prune": false,
"state": "active",
"targetNamespace": "datadog",
"transitioning": "no",
"transitioningMessage": "",
"type": "app",
"uuid": "8f7a38e7-7b5b-11ea-8ce6-0242ac110002",
"valuesYaml": "---\n  defaultImage: true\n  image: \n    repository: \"ranchercharts/datadog-agent\"\n    tag: \"6.13.0\"\n  clusterAgent: \n    image: \n      repository: \"ranchercharts/datadog-cluster-agent\"\n      tag: \"1.3.2\"\n    enabled: false\n    metricsProvider: \n      enabled: true\n  kube-state-metrics: \n    image: \n      repository: \"ranchercharts/coreos-kube-state-metrics\"\n      tag: \"v1.7.2\"\n  datadog: \n    appKey: \"\"\n    appKeyExistingSecret: \"\"\n    apiKey: \"soumyatest\"\n    apiKeyExistingSecret: \"\"\n    site: \"datadoghq.com\"\n    dd_url: \"\"\n    logLevel: \"warn\"\n    tags: \"\"\n    useCriSocketVolume: true\n    criSocketPath: \"\"\n    nonLocalTraffic: false\n    collectEvents: false\n    podLabelsAsTags: \"\"\n    podAnnotationsAsTags: \"\"\n    nodeLabelsAsTags: \"\"\n    httpProxy: \"\"\n    httpsProxy: \"\"\n    noProxy: \"\"\n    apmEnabled: false\n    apmNonLocalTraffic: false\n    processAgentEnabled: false\n    logsEnabled: false\n  daemonset: \n    rkeETCDControlPlane: true\n    useHostNetwork: false\n    useHostPort: false\n    useHostPID: false\n  kubeStateMetrics: \n    enabled: true"

```
Upgraded the app through CLI and  upgrade failed
```
./rancher app upgrade --values values.yml datadog 2.1.2
```
The API displayed the values in answers section:
```
{
"actions": {
"rollback": "…/v3/project/c-v5l2s:p-5p794/apps/p-5p794:datadog?action=rollback",
"upgrade": "…/v3/project/c-v5l2s:p-5p794/apps/p-5p794:datadog?action=upgrade"
},
"annotations": {
"lifecycle.cattle.io/create.helm-controller_c-v5l2s": "true"
},
"answers": {
"clusterAgent.enabled": "false",
"clusterAgent.image.repository": "ranchercharts/datadog-cluster-agent",
"clusterAgent.image.tag": "1.3.2",
"clusterAgent.metricsProvider.enabled": "true",
"daemonset.rkeETCDControlPlane": "true",
"daemonset.useHostNetwork": "false",
"daemonset.useHostPID": "false",
"daemonset.useHostPort": "false",
"datadog.apiKey": "soumyatest",
"datadog.apiKeyExistingSecret": "",
"datadog.apmEnabled": "false",
"datadog.apmNonLocalTraffic": "false",
"datadog.appKey": "",
"datadog.appKeyExistingSecret": "",
"datadog.collectEvents": "false",
"datadog.criSocketPath": "",
"datadog.dd_url": "",
"datadog.httpProxy": "",
"datadog.httpsProxy": "",
"datadog.logLevel": "warn",
"datadog.logsEnabled": "false",
"datadog.noProxy": "",
"datadog.nodeLabelsAsTags": "",
"datadog.nonLocalTraffic": "false",
"datadog.podAnnotationsAsTags": "",
"datadog.podLabelsAsTags": "",
"datadog.processAgentEnabled": "false",
"datadog.site": "datadoghq.com",
"datadog.tags": "",
"datadog.useCriSocketVolume": "true",
"defaultImage": "true",
"image.repository": "ranchercharts/datadog-agent",
"image.tag": "6.13.0",
"kube-state-metrics.image.repository": "ranchercharts/coreos-kube-state-metrics",
"kube-state-metrics.image.tag": "v1.7.2",
"kubeStateMetrics.enabled": "true"
}
```
